API/src/main/java/de/brandwatch/minianalytics/api/controller/MentionController.java

```diff
@@ -0,0 +1,62 @@
+package de.brandwatch.minianalytics.api.controller;
+
+
+import com.google.gson.Gson;
+import de.brandwatch.minianalytics.api.solr.model.Mention;
+import de.brandwatch.minianalytics.api.solr.repository.MentionRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+
+@RestController
+public class MentionController {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestController.class);
+
+    private static final Gson gson = new Gson();
+
+    private final MentionRepository mentionRepository;
+
+    @Autowired
+    public MentionController(MentionRepository mentionRepository) {
+        this.mentionRepository = mentionRepository;
+    }
+
+    @GetMapping(value = "/mentions/{queryID}")
+    public ResponseEntity findMentionsFromQuery(@PathVariable String queryID, @RequestParam(value = "date", defaultValue = "") String date){
+
+        logger.info("GET /mentions/" + queryID);
+
+        List<Mention> mentions = mentionRepository.findByQueryID(Long.parseLong(queryID));
+
+        if(!date.equals("")){
```

Comment:
I would say the stuff below belongs on a service layer, but honestly I am not 100% sure what you are doing there. Maybe you clan clarify, the whole conversion and those several date formats are not clear to me. It's also not clear to me why you fetch mentions twice.

What you should be doing if you want date as a filter (which is definitely desirable): Have `startDate` and `endDate` request params that both have sensible defaults, let's say `endDate` is now and `startDate` is now-24hrs, or maybe better `endDate`-24hrs, and are both not required. You need some validation code that belongs into the controller and validates that for example `endDate` is after `startDate`.

Then pass all parameters to a service layer (either the defaults or what was passed from the client) and use the service layer to convert things to Solr format (i.e. the `dateBounds`) and then pass it on to the repository.

Reply:
I already moved this to the service layer (https://github.com/Max-Leopold/mini-analytics/pull/34). 

I started with just an option to set up the start date. The end date is always set to now. The next step would be to also be able to set an end date.

Reply:
Wrong PR sorry. This one is right: https://github.com/Max-Leopold/mini-analytics/pull/35/commits/fb2498c636948035c9ed4192123f5ebcdf12e176

Reply:
I see. Then it's ok for this, but please consider my comments for that other one. You can make the changes there.

Reply:
Changed discussed things here: https://github.com/Max-Leopold/mini-analytics/commit/0ce83522a901c514fb56ead9aad5ff21c1c12f60

Reply:
Which PR does that commit belong to?

Reply:
I don't actually know why, but it apparently belongs to this one https://github.com/Max-Leopold/mini-analytics/pull/36. Don't know why I put it in there tho...

Reply:
I could rebase this branch onto #36. Then all changes done there regarding architecture would also be in this PR.

Reply:
Not sure, I think first you need to be sure about what's going on with 36. It seems to do a lot more than adding tests as the title claims.

Reply:
Many of the commits are form #35. Once that one is merges, I can rebase the #36 and it will do much less. It does add Controller Tests and remove them in a later commit, so the title is wrong. Problem is, the #36 doesn't serve a clear purpose now. It is just an extension from #35

Reply:
That still doesn't make sense. https://github.com/Max-Leopold/mini-analytics/commit/0ce83522a901c514fb56ead9aad5ff21c1c12f60 is a logical change, but both PRs claim to be only about testing.

Reply:
Yes https://github.com/Max-Leopold/mini-analytics/commit/0ce83522a901c514fb56ead9aad5ff21c1c12f60 should have its own PR, but had to be on this branch with #35 because the MentionService was added there

Reply:
So I fixed this now. This is how:

1. I added a new branch `Add-MentionService` #41. This PR adds the MentionService and makes the changes discussed here: https://github.com/Max-Leopold/mini-analytics/pull/22#discussion_r320258648
2. I rebased #35 to sit on top of #41. In two commits the two tests for MentionService and QueryService are added. 
3. I removed #36 and the branch that belonged to #36. Simply because that branch didn't do anything. It first added ControllerTests and later removed them.

I think this should be clearer now.

Reply:
I don't have a Validator yet. But that will be added.

Reply:
Now I also have a Validator https://github.com/Max-Leopold/mini-analytics/pull/41/commits/c7c15dce58a1d34094534aebbfc8b27f3e99ad8d. I put it in the service layer, because it has some business logic inside.